python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,553 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Distill wrapper.
+"""
+
+import torch as th
+from torch.nn.parallel import DistributedDataParallel
+import os
+import json
+import pandas as pd
+
+from ..model.gnn import do_full_graph_inference
+from ..utils import sys_tracker
+from .utils import remap_embeddings
+
+from ..model.gnn_distill import GSDistilledModel
+from ..dataloading import DataloaderGenerator, DataProvider
+
+class GSdistiller():
+    """ GNN distiller.
+
+    Parameters
+    ----------
+    model : GSgnnLinkPredictionModelBase
+        The GNN model for GNN embeddings inference.
+    rank : int
+        The rank.
+    config : GSConfig
+        Configs for GNN distillation
+    """
+    def __init__(self, model, rank, config):
```

Comment:
we don't want to pass the config object to the distiller object. please see how we implement other components, such as trainer, inferer and everything else.